### PR TITLE
Share by dynamic links service 

### PR DESCRIPTION
**Pull request by MahmoudHafezDarwish**

### CODE REVIEW

**Review by HalaSalim77 — APPROVED** [review]
🥇

File: lib/dynamic_links_service.dart [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        await FirebaseDynamicLinks.instance.buildShortLink(
+      parameters,
+    );
+    final Uri shortUrl = shortDynamicLink.shortUrl;
+
+    return shortUrl.toString();
+  }
+
+  static void initDynamicLinks() async {
+    await Future.delayed(const Duration(seconds: 2));
```
Comment (salahamassi) [inline lib/dynamic_links_service.dart:33]: getInitialLink rename it

File: lib/dynamic_links_service.dart [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    final PendingDynamicLinkData? data =
```
Comment (salahamassi) [inline lib/dynamic_links_service.dart:34]: don't declare types

File: lib/dynamic_links_service.dart [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        await FirebaseDynamicLinks.instance.getInitialLink();
+    if (appRouter.stack.isEmpty) {
```
Comment (salahamassi) [inline lib/dynamic_links_service.dart:36]: useless if

File: lib/dynamic_links_service.dart [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      _handleDynamicLink(data);
+    }
+
+    FirebaseDynamicLinks.instance.onLink.listen((PendingDynamicLinkData? data) {
```
Comment (salahamassi) [inline lib/dynamic_links_service.dart:40]: listen to dynamic links

Comment (salahamassi) [reply lib/dynamic_links_service.dart:40]: listen to return stream 
must capture on variable and cancel it on dispose

Comment (salahamassi) [reply lib/dynamic_links_service.dart:40]: APP BLOC

File: lib/dynamic_links_service.dart [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      return;
+    }).onError((error) {});
+  }
+
+  static Future<void> _handleDynamicLink(PendingDynamicLinkData? data) async {
+    if (data == null) {
+      return;
+    }
+    final Uri deepLink = data.link;
+    if (deepLink.queryParameters['type'] == 'post') {
```
Comment (salahamassi) [inline lib/dynamic_links_service.dart:51]: 100 DYNAMIC LINK

File: lib/dynamic_links_service.dart (hunk 1 above)
Comment (MahmoudHafezDarwish) [reply lib/dynamic_links_service.dart:33]: done

File: lib/dynamic_links_service.dart (hunk 2 above)
Comment (MahmoudHafezDarwish) [reply lib/dynamic_links_service.dart:34]: done

File: lib/dynamic_links_service.dart (hunk 3 above)
Comment (MahmoudHafezDarwish) [reply lib/dynamic_links_service.dart:36]: done

**Review by Mamoun2020 — COMMENTED** [review]
Great Job, Go ahead 🤘🔥

File: lib/dynamic_links_service.dart (hunk 5 above)
Comment (MahmoudHafezDarwish) [reply lib/dynamic_links_service.dart]: done, please check it

**Pull request closed without merging**